github/providers.tf

```diff
@@ -0,0 +1,84 @@
+locals {
+  gh_token         = try(local.secrets[var.gh_token_secret_name], "")
+  gh_token_enabled = length(local.gh_token) > 0 ? true : false
+
+  gh_app_auth_id_enabled              = length(var.gh_app_auth_id) > 0 ? true : false
+  gh_app_auth_installation_id_enabled = length(var.gh_app_auth_installation_id) > 0 ? true : false
+  gh_app_auth_pem_file                = try(local.secrets["${var.gh_app_auth_pem_file_secret_name}"], "")
+  gh_app_auth_pem_file_enabled        = length(local.gh_app_auth_pem_file) > 0 ? true : false
+  gh_app_auth_enabled                 = alltrue([local.gh_app_auth_id_enabled, local.gh_app_auth_installation_id_enabled, local.gh_app_auth_pem_file_enabled])
+}
+
+# The GitHub provider offers multiple ways to authenticate with GitHub API:
+# - GitHub CLI
+# - OAuth / Personal Access Token
+# - GitHub App Installation
+#
+# When none configuration is provided, the provider can only access resources available anonymously.
+provider "github" {
+  base_url = var.gh_base_url
+  owner    = var.gh_owner
+
+  token = local.gh_token_enabled ? local.gh_token : null
+
+  dynamic "app_auth" {
+    for_each = local.gh_app_auth_enabled ? ["app_auth"] : []
+    content {
+      id              = var.gh_app_auth_id
+      installation_id = var.gh_app_auth_installation_id
+      pem_file        = local.gh_app_auth_pem_file
+    }
+  }
+}
```

Comment:
This can all be handled in environment variables and that sort of configuration will remain consistent in Spacelift, then we don't have to do the auth method selection logic in locals, but there's also merits to it being explicit in code. Especially because we don't use a `chamber` sort of environment variable management in local dev loop.

Reply:
We rarely use env variables for this type of configuration, so I leaned towards the approach that keeps GH configuration in one place without a need to set up Spacelift stacks (in our case). With this implementation, a token can still be consumed as an env variable if not set via SOPS. 
For app_auth we could work on more complex logic in the future if needed.